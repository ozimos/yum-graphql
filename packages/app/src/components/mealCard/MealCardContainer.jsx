import React from 'react';
import PropTypes from 'prop-types';


const MealCardContainer = ({ MealCard, meals, ...props }) =>
  (
    <div className={props.addClass ? `${props.addClass} gallery` : 'gallery'}>
      {meals.map(meal => (<MealCard
        key={meal.id}
        {...meal}
        {...props}
      />))
        }
    </div>
  );

MealCardContainer.defaultProps = {
  addClass: ''
};

MealCardContainer.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  MealCard: PropTypes.func.isRequired,
  addClass: PropTypes.string,
};

export default MealCardContainer;

