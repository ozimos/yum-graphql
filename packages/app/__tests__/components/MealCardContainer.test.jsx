/* global React:false, shallow:false toJson:false */

import MealCardContainer from '../../src/components/mealCard/MealCardContainer';
import { Meals } from '../__mocks__/mealDataMock';

const props = {
  meals: Meals,
  MealCard: jest.fn(),
};

describe('MealCardContainer Component', () => {
  const setup = () => shallow(<MealCardContainer {...props} />);

  it('renders correctly', () => {
    let wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
    const propsWithExtraClass = { ...props, addClass: 'string' };
    const setupWithExtraClass = () => shallow(<MealCardContainer
      {...propsWithExtraClass}
    />);
    wrapper = setupWithExtraClass();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
