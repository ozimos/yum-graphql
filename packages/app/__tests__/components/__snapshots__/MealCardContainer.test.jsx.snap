// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCardContainer Component renders correctly 1`] = `
<div
  className="gallery"
>
  <mockConstructor
    addClass=""
    description="nice"
    id="meal1"
    imageUrl="nice"
    key="meal1"
    price={1200}
    title="Eba and Egusi"
    userId="user1"
  />
  <mockConstructor
    addClass=""
    description="nice"
    id="meal2"
    imageUrl="nice"
    key="meal2"
    price={2000}
    title="Rice and Stew"
    userId="user1"
  />
</div>
`;

exports[`MealCardContainer Component renders correctly 2`] = `
<div
  className="string gallery"
>
  <mockConstructor
    addClass="string"
    description="nice"
    id="meal1"
    imageUrl="nice"
    key="meal1"
    price={1200}
    title="Eba and Egusi"
    userId="user1"
  />
  <mockConstructor
    addClass="string"
    description="nice"
    id="meal2"
    imageUrl="nice"
    key="meal2"
    price={2000}
    title="Rice and Stew"
    userId="user1"
  />
</div>
`;
